torch/nn/modules/container.py

```diff
@@ -125,17 +125,18 @@ def append(self, module):
         return self
 
     def extend(self, modules):
-        r"""Appends modules from a Python list at the end.
+        r"""Appends modules from a Python iterable to the end of the list.
 
         Arguments:
-            modules (list): list of modules to append
+            modules (iterable): iterable of modules to append
         """
-        if not isinstance(modules, list):
+        if not isinstance(modules, Iterable) or isinstance(modules, str):
             raise TypeError("ModuleList.extend should be called with a "
-                            "list, but got " + type(modules).__name__)
+                            "non-string iterable, but got " +
+                            type(modules).__name__)
         offset = len(self)
         for i, module in enumerate(modules):
-            self.add_module(str(offset + i), module)
+            self.add_module(str(len(self)), module)
```

Comment:
Any reason to change `offset + i` to `len(self)`? Now both `offset` and `i` are unused. BTW the same pattern is still used below.